Tests/SonarScanner.MSBuild.PreProcessor.Test/ProcessedArgsTests.cs

```diff
@@ -152,6 +152,100 @@ public void ProcArgs_CmdLinePropertiesOverrideFileSettings()
         }
 
+        [TestMethod]
+        public void ProcArgs_HostUrl_SonarcloudUrl_HostUrlSet()
+        {
+            var sut = new ProcessedArgs("key", "name", "version", "organization", false, new ListPropertiesProvider([
+                new Property(SonarProperties.HostUrl, "http://host"),
+            ]), EmptyPropertyProvider.Instance, EmptyPropertyProvider.Instance, logger);
+            sut.SonarServer.Should().BeOfType<SonarQubeServer>().Which.ServerUrl.Should().Be("http://host");
+            logger.Warnings.Should().BeEmpty();
+            logger.Errors.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void ProcArgs_HostUrl_SonarcloudUrl_SonclourUrlSet()
+        {
+            var sut = new ProcessedArgs("key", "name", "version", "organization", false, new ListPropertiesProvider([
+                new Property(SonarProperties.SonarcloudUrl, "https://sonarcloud.proxy"),
+                ]), EmptyPropertyProvider.Instance, EmptyPropertyProvider.Instance, logger);
+            sut.SonarServer.Should().BeOfType<SonarCloudServer>().Which.ServerUrl.Should().Be("https://sonarcloud.proxy");
+            logger.Warnings.Should().BeEmpty();
+            logger.Errors.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void ProcArgs_HostUrl_SonarcloudUrl_HostUrlAndSonarcloudUrlAreIdentical()
+        {
+            var sut = new ProcessedArgs("key", "name", "version", "organization", false, new ListPropertiesProvider([
+                new Property(SonarProperties.HostUrl, "https://sonarcloud.proxy"),
+                new Property(SonarProperties.SonarcloudUrl, "https://sonarcloud.proxy"),
+            ]), EmptyPropertyProvider.Instance, EmptyPropertyProvider.Instance, logger);
+            sut.SonarServer.Should().BeOfType<SonarCloudServer>().Which.ServerUrl.Should().Be("https://sonarcloud.proxy");
+            logger.AssertWarningLogged("The sonar.host.url value and sonar.scanner.sonarcloudUrl are both set. Set only sonar.scanner.sonarcloudUrl.");
+            logger.Errors.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void ProcArgs_HostUrl_SonarcloudUrl_HostUrlAndSonarcloudUrlDiffer()
+        {
+            var sut = new ProcessedArgs("key", "name", "version", "organization", false, new ListPropertiesProvider([
+                new Property(SonarProperties.HostUrl, "https://someHost.com"),
+                new Property(SonarProperties.SonarcloudUrl, "https://someOtherHost.org"),
+            ]), EmptyPropertyProvider.Instance, EmptyPropertyProvider.Instance, logger);
+            sut.SonarServer.Should().BeNull();
+            logger.Warnings.Should().BeEmpty();
+            logger.AssertErrorLogged("The sonar.host.url value and sonar.scanner.sonarcloudUrl are both set and are different. " +
+                "Set either sonar.host.url for SonarQube and sonar.scanner.sonarcloudUrl for SonarCloud.");
```

Comment:
Shouldn't be an `or` instead of `and` this this case?
```suggestion
                "Please set either `sonar.host.url` for SonarQube or `sonar.scanner.sonarcloudUrl` for SonarCloud.");
```

Reply:
👍 but are you sure about the ` ?

Reply:
It will not format the code but I think it will show better the property name. We could also use single quotes.